evidence/errors.go

```diff
@@ -0,0 +1,20 @@
+package evidence
+
+import (
+	"fmt"
+)
+
+type errInvalidEvidence struct {
```

Comment:
Why have a private type with public field(s)? Generally, typed errors should be public so external logic can check cause if need be, no?

Reply:
Yeah I changed the type to public as it was being returned by a public function